docs/dev/docs.md

```diff
@@ -0,0 +1,126 @@
+# Developing documentation
```

Comment:
Do we want to mention, somewhere in this document, that we also have some other off-site docs for the "Glean as a product" thing? I mean [these docs](https://docs.telemetry.mozilla.org/concepts/glean/glean.html).

Reply:
Good point.